src/main/src/main/java/org/geoserver/catalog/impl/StyleInfoImpl.java

```diff
@@ -97,6 +97,9 @@ public void setFilename(String filename) {
     }
 
     public Style getStyle() throws IOException {
+        // for cap doc return remote name
+        if (metadata != null)
+            if (metadata.containsKey("isRemote")) return CascadedLayerInfoImpl.getStyleInfo(this);
         return catalog.getResourcePool().getStyle(this);
```

Comment:
The two lines of code above are a unreadable for me, and the comment doesn't explain much.

Reply:
Allow me to explain

This is to make Cascaded layer advertise remote styles in GetCapability request. Since we have implemented the GetStyles and GetDefaultStyle in CascadedLayer, the code is trying to look for the Style (which is a remote style) in the catalog, but we never saved the Remote styles in Catalog (because they exist on another server and we don't want to them show up as local resources). So here we are looking for a flag in metadata map of Style to indicate that don't look in the catalog for this Style, just take it as it is.

Reply:
I will fix the comments

Reply:
kk.